homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,142 @@
+"""Assist satellite entity."""
+
+from abc import abstractmethod
+from collections.abc import AsyncIterable
+import time
+from typing import Final
+
+from homeassistant.components import stt
+from homeassistant.components.assist_pipeline import (
+    AudioSettings,
+    PipelineEvent,
+    PipelineEventType,
+    PipelineStage,
+    async_pipeline_from_audio_stream,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import Context
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.util import ulid
+
+from .models import AssistSatelliteState, SatelliteConfig
+
+_CONVERSATION_TIMEOUT_SEC: Final = 5 * 60  # 5 minutes
+
+
+class AssistSatelliteEntity(entity.Entity):
+    """Entity encapsulating the state and functionality of an Assist satellite."""
+
+    entity_description = EntityDescription(
+        key="assist_satellite",
+        translation_key="assist_satellite",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_should_poll = False
+    _attr_state: AssistSatelliteState | None = (
+        AssistSatelliteState.WAITING_FOR_WAKE_WORD
+    )
+
+    _satellite_config = SatelliteConfig()
+
+    _conversation_id: str | None = None
+    _conversation_id_time: float | None = None
+
+    _run_has_tts: bool = False
+
+    async def _async_accept_pipeline_from_satellite(
+        self,
+        context: Context,
```

Comment:
Instead of passing context, consider using https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L918-L919